tests/storage/restore.go

```diff
@@ -1201,6 +1201,49 @@ var _ = Describe(SIG("VirtualMachineRestore Tests", func() {
 				libvmops.StartVirtualMachine(vm)
 			})
 
+			It("restore should fail immediately with targetReadinessPolicy FailImmediate and status should not change after stopping VM", func() {
```

Comment:
Can't we completely unit test this entire scenario? kind of a bummer to be creating real VMs

Reply:
yeah I didnt add it at first, and then I thought it can be good to make sure this works with the entire workflow. But yeah you are right Ill make sure its fully covered with the unit tests and remove it.